src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -722,6 +723,58 @@ CreateOffer::applyGuts (ApplyView& view, ApplyView& view_cancel)
 
     if (result == tesSUCCESS)
     {
+        // If a tick size applies, round the offer to the tick size
+        auto const& uPaysIssuerID = saTakerPays.getIssuer ();
+        auto const& uGetsIssuerID = saTakerGets.getIssuer ();
+
+        std::uint8_t uTickSize = Quality::maxTickSize;
+        if (!isXRP (uPaysIssuerID))
+        {
+            auto const sle =
+                view.read(keylet::account(uPaysIssuerID));
+            if (sle && sle->isFieldPresent (sfTickSize))
+                uTickSize = std::min (uTickSize,
+                    (*sle)[sfTickSize]);
+        }
+        if (!isXRP (uGetsIssuerID))
+        {
+            auto const sle =
+                view.read(keylet::account(uGetsIssuerID));
+            if (sle && sle->isFieldPresent (sfTickSize))
+                uTickSize = std::min (uTickSize,
+                    (*sle)[sfTickSize]);
+        }
+        if (uTickSize < Quality::maxTickSize)
+        {
+            auto const rate =
+                Quality{saTakerPays, saTakerGets}.round
```

Comment:
Not specific to this PR, but I'm confused on the order of arguments to `Quality` for offers.   In https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/Quality.h#L39-L40, it states `takerPays` is always  the 'in' argument to a `TAmount`.  For the constructor call here, based on https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/Quality.h#L143-L146 it looks like `saTakerPays` ends up as the `out` amount.